packages/zoe/contracts/helpers/userFlow.js

```diff
@@ -10,15 +10,20 @@ const hasAssays = (assays, newPayoutRules) =>
   assays.every((assay, i) => assay === newPayoutRules[i].units.label.assay);
 
 export const makeHelpers = (zoe, assays) => {
+  const unitOpsArray = zoe.getUnitOpsForAssays(assays);
   const helpers = harden({
     getActiveOffers: handles =>
       zoe.getOffers(zoe.getOfferStatuses(handles).active),
     rejectOffer: (inviteHandle, msg = defaultRejectMsg) => {
       zoe.complete(harden([inviteHandle]));
       throw new Error(msg);
     },
+    areAssetsEqualAtIndex: (index, leftHandle, rightHandle) =>
+      unitOpsArray[index].equals(
+        zoe.getOffer(leftHandle).payoutRules[index].units,
+        zoe.getOffer(rightHandle).payoutRules[index].units,
+      ),
     canTradeWith: inviteHandles => {
-      const unitOpsArray = zoe.getUnitOpsForAssays(assays);
       const { payoutRules: leftPayoutRules } = zoe.getOffer(inviteHandles[0]);
```

Comment:
`canTradeWith`'s code really is specific to exactly two handles. The question it asks doesn't generalize anyway. While you could add an `insist` that `inviteHandles` is length two, I think it'll generally be clearer to just have two handle arguments. May as well be `leftHandle` and `rightHandle` to parallel `areAssetsEqual`.

Reply:
good point